Server creation fails with "OS (linux) not supported" error even though template doesn't list an OS requirement

I'm trying to add some servers to a local PufferPanel dev environment so I can dig into #1149 a bit more. For now, I'm working on Linux (Debian in WSL2, to be specific) so I can experiment with behavior there and see how it differs from the behavior I observe on Windows. However, in the latest Git master, when I try create a server from one of the stock templates, creation fails with this error:

```
[GIN] 2022/04/18 - 18:27:30 | 200 |       439.7µs |             ::1 | GET      "/api/templates/minecraft-vanilla"
[ERROR] 2022/04/18 18:27:43 OS (linux) not supported. Supported OS:
[GIN] 2022/04/18 - 18:27:43 | 400 |      260.23µs |                 | PUT      "/daemon/server/bb73e45c"
[ERROR] 2022/04/18 18:27:43 Unexpected response from daemon: 400
[ERROR] 2022/04/18 18:27:43 unknown error
```

I think it's trying to apply [this requirements check](https://github.com/PufferPanel/PufferPanel/blob/42fd3c02e625da978d7d1d7bcad28a5578a72fcb/server.go#L101-L113) even though my template doesn't have a `requirements` section populated.

--- a/server.go
+++ b/server.go
@@ -15,11 +15,12 @@ package pufferpanel
 
 import (
 	"context"
-	"github.com/docker/docker/client"
 	"os/exec"
 	"runtime"
 	"strings"
 	"time"
+
+	"github.com/docker/docker/client"
 )
 
 type Server struct {
@@ -185,6 +186,9 @@ func (s Server) DataToMap() map[string]interface{} {
 }
 
 func parseRequirementRow(str string) []string {
+	if str == "" {
+		return []string{}
+	}
 	d := strings.Split(str, "||")
 	for k, v := range d {
 		d[k] = strings.TrimSpace(v)

--- a/server_test.go
+++ b/server_test.go
@@ -1,0 +1,26 @@
+package pufferpanel
+
+import (
+	"reflect"
+	"testing"
+)
+
+func TestParseRequirementRow(t *testing.T) {
+	tests := []struct {
+		str  string
+		want []string
+	}{
+		{str: "", want: []string{}},
+		{str: "linux", want: []string{"linux"}},
+		{str: "  linux   ", want: []string{"linux"}},
+		{str: "linux||windows", want: []string{"linux", "windows"}},
+		{str: " linux    || windows  ", want: []string{"linux", "windows"}},
+	}
+
+	for _, tt := range tests {
+		got := parseRequirementRow(tt.str)
+		if !reflect.DeepEqual(got, tt.want) {
+			t.Errorf("parseRequirementRow(%#v) = %v, want %v", tt.str, got, tt.want)
+		}
+	}
+}

